pkg/app/api/grpcapi/web_api.go

```diff
@@ -194,6 +197,83 @@ func (a *WebAPI) DeleteEnvironment(ctx context.Context, req *webservice.DeleteEn
 	if err := a.validateEnvBelongsToProject(ctx, req.EnvironmentId, claims.Role.ProjectId); err != nil {
 		return nil, err
 	}
+	// Check if no Piped has permission to the given environment.
+	pipeds, err := a.pipedStore.ListPipeds(ctx, datastore.ListOptions{
+		Filters: []datastore.ListFilter{
+			{
+				Field:    "EnvIds",
+				Operator: datastore.OperatorIn,
+				Value:    req.EnvironmentId,
+			},
```

Comment:
@nakabonne This `EnvIds` field does not exist as a column (as both Stored or Virtual) in Piped table schema for MySQL datastore, so I think this will cause unable to perform query problem if MySQL datastore is used. 👀 

Another thing that concerned me is this `OperatorIn`. It's more like `BelongsTo` to me.
Maybe just my misunderstanding in case of Firestore datastore is used 🤔  But in case of MySQL, we will generate a query that looks like this for IN operator

https://github.com/pipe-cd/pipe/blob/master/pkg/datastore/mysql/query_test.go#L261-L274

Reply:
> this will cause unable to perform query problem if MySQL datastore is used. 👀 We can make an issue for that and resolve it later.

Got it, I'll open up a new issue for that.

>Another thing that concerned me is this OperatorIn. It's more like kind of CONTAINS to me.

for sure. I'll look up it a bit more.